src/components/BlogPost.js

```diff
@@ -1,45 +1,40 @@
-import React from 'react'
-import styled from 'styled-components'
-
-const BlogPostWrapper = styled.div`
-  border: 3px solid #000;
-  width: 50%;
-`
-
-const BlogPostDate = styled.h1`
-  color: #7cc2fd;
-`
-
-const BlogPostTitle = styled.h2`
-  color: #f268ae;
+import PropTypes from "prop-types"
+import React from "react"
+import styled from "styled-components"
+
+import Date from "./Date"
+import Image from "./Image"
+import Paragraph from "./Paragraph"
+import Tag from "./Tag"
+import Title from "./Title"
+
+const BlogPostWrapper = styled.section`
+  margin: 20px;
+  flex-direction: row;
 `
 
-const BlogPostTags = styled.li`
-  color: #dcce76;
-  list-style-type: none;
-`
-
-const BlogPostText = styled.p`
-  font-size: 14px;
+const ImageWrapper = styled.div`
+  display: inline-block;
 `
 
 const BlogPost = ({ post }) => (
   <BlogPostWrapper>
-    <BlogPostDate>
-      {post.publishedDate}
-    </BlogPostDate>
-    <BlogPostTags>
-      {post.tags}
-    </BlogPostTags>
-    <BlogPostTitle>
-      {post.title}
-    </BlogPostTitle>
-
-    <img src={post.media[0].file.url} alt="" />
-    <BlogPostText>
-      {post.description.internal.content}
-    </BlogPostText>
+    <Date date={post.publishedDate} />
+    {post.tags.map(tag => (
+      <Tag key={tag} tag={tag} />
+    ))}
+    <Title title={post.title} />
+    <ImageWrapper>
+      {post.media.map((image, index) => (
```

Comment:
small nit: instead of using an index for key, you can just use `image.file.url` since that value is always unique. I believe eslint raises a warning about this.

Reply:
Yes, I posted a long comment here about indexes.  Missed that I had used one here, though.  Thanks for the catch!